sandbox/modules/identity/src/main/java/org/opensearch/identity/rest/api/IdentityAbstractRestApiAction.java

```diff
@@ -0,0 +1,565 @@
+        final String id = cType.toLCString();
+
+        configuration.removeStatic();
+
+        try {
+            client.index(
+                ir.id(id)
+                    .setRefreshPolicy(RefreshPolicy.IMMEDIATE)
+                    .setIfSeqNo(configuration.getSeqNo())
+                    .setIfPrimaryTerm(configuration.getPrimaryTerm())
+                    .source(id, XContentHelper.toXContent(configuration, XContentType.JSON, false)),
+                new ConfigUpdatingActionListener<>(new String[] { id }, client, actionListener)
+            );
+        } catch (IOException e) {
+            throw ExceptionsHelper.convertToOpenSearchException(e);
+        }
+    }
+
+    protected static class ConfigUpdatingActionListener<Response> implements ActionListener<Response> {
+        private final String[] cTypes;
+        private final Client client;
+        private final ActionListener<Response> delegate;
+
+        public ConfigUpdatingActionListener(String[] cTypes, Client client, ActionListener<Response> delegate) {
+            this.cTypes = Objects.requireNonNull(cTypes, "cTypes must not be null");
+            this.client = Objects.requireNonNull(client, "client must not be null");
+            this.delegate = Objects.requireNonNull(delegate, "delegate must not be null");
+        }
+
+        @Override
+        public void onResponse(Response response) {
+
+            final ConfigUpdateRequest cur = new ConfigUpdateRequest(cTypes);
+
+            client.execute(ConfigUpdateAction.INSTANCE, cur, new ActionListener<>() {
+                @Override
+                public void onResponse(final ConfigUpdateResponse ur) {
+                    if (ur.hasFailures()) {
+                        delegate.onFailure(ur.failures().get(0));
+                        return;
+                    }
+                    delegate.onResponse(response);
+                }
+
+                @Override
+                public void onFailure(final Exception e) {
+                    delegate.onFailure(e);
+                }
+            });
+
+        }
+
+        @Override
+        public void onFailure(Exception e) {
+            delegate.onFailure(e);
+        }
+
+    }
+
+    @Override
+    protected final RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+
+        // consume all parameters first so we can return a correct HTTP status,
+        // not 400
+        consumeParameters(request);
+
+        // check if .opendistro_security index has been initialized
+        if (!ensureIndexExists()) {
+            return channel -> internalErrorResponse(channel, ErrorType.SECURITY_NOT_INITIALIZED.getMessage());
+        }
+
+        // check if request is authorized
+        String authError = restApiPrivilegesEvaluator.checkAccessPermissions(request, getEndpoint());
+
+        final User user = (User) threadPool.getThreadContext().getTransient(ConfigConstants.IDENTITY_USER);
```

Comment:
This is where we will want to use the `Identity.getAuthManager().getSubject()` and abstract this logic out.